### PR TITLE
4.3 Release Notes and Changelog

### PR DESCRIPTION
**Pull request by mtopolnik**

### CODE REVIEW

File: hazelcast-jet-distribution/src/root/release_notes.txt [hunk 1]
```diff
@@ -23,6 +23,42 @@ To expand an issue number into a full GitHub link, use this template:
 
 https://github.com/hazelcast/hazelcast-jet/issues/<issue number>
 
+
+** Hazelcast Jet 4.3 **
```
Comment (cangencer) [inline hazelcast-jet-distribution/src/root/release_notes.txt:27]: Are these all?

* IMDG version is updated
* I can see many other bugs fixed: https://github.com/hazelcast/hazelcast-jet/pulls?q=is%3Apr+milestone%3A4.3+is%3Aclosed+label%3Adefect but only 1 is mentioned here?

File: hazelcast-jet-distribution/src/root/release_notes.txt [hunk 2]
```diff
@@ -23,6 +23,42 @@ To expand an issue number into a full GitHub link, use this template:
...
+1. New Features
+
+[pipeline-api] [014] @MohamedMandouh implemented distributed sorting: `BatchStage.sort()` (#2469, #2544)
+[core] [012] Added `JobConfig.suspendOnFailure`: suspend a job on exception instead of cancelling it (#2411)
+
+
+2. Enhancements
+
+[core] @guenter-hesse contributed a test source to benchmark Jet's throughput and latency (#2382)
+[core] [013] Avoid problem where low event rate would stall progress of a processing time-based pipeline (#2485, #2514)
```
Comment (jbartok) [inline hazelcast-jet-distribution/src/root/release_notes.txt:46]: Maybe mention the addition of configurable reconnect strategies for CDC sources?

File: hazelcast-jet-distribution/src/root/release_notes.txt [hunk 3]
```diff
@@ -23,6 +23,42 @@ To expand an issue number into a full GitHub link, use this template:
...
+
+
+3. Fixes
+
+[core] Fixed a bug where a non-keyed aggregating stage would produce no output when no input (#2560, #2567)
+
```
Comment (jbartok) [inline hazelcast-jet-distribution/src/root/release_notes.txt:52]: Maybe mention the integration of a Jackson-jr bug-fix, which was affecting our JSON-to-POJO mapping, specifically in CDC (https://github.com/FasterXML/jackson-jr/issues/73).

Also, integration of a Debezium fix which solved the problem of losing data on snapshotting in CDC sources for Postgres (https://issues.redhat.com/browse/DBZ-2288).

**Review by jbartok — CHANGES_REQUESTED** [review]

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder-docs-no-build` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/295/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/295/jet-oss-pr-builder-docs-no-build-295.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
There are unexpected files.
Unexpected file: NOTICE
</details>
</pre>

**Review by jbartok — APPROVED** [review]

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder-docs-no-build` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/297/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/297/jet-oss-pr-builder-docs-no-build-297.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
There are unexpected files.
Unexpected file: NOTICE
</details>
</pre>

File: hazelcast-jet-distribution/src/root/release_notes.txt [hunk 4]
```diff
@@ -23,6 +23,70 @@ To expand an issue number into a full GitHub link, use this template:
...
+  `newJob()` (#2481, #2484)
+[core] Fix a race between snapshotting and restarting (#2487, #2503)
+[core] Fixed a race where `getJobStatus()` would report `RUNNING` even though it was actually
+  `COMPLETING`. (#2507)
+[core] Fixed an issue where a `DONE_ITEM` could get lost due to connection failure, preventing
+  the job from completing (#2158, #2532)
+[core] Fixed a job failure related to the coordinator member failing (#2461, #2546)
+[core] Fixed a job failure related to a member reconnecting (#2542, #2547)
+[core] Improved robustness related to Jet's internal `IMap` operations (#2533, #2550)
+[cdc] Upgraded Jackson jr dep, solving a null handling issue (#2459)
```
Comment (jbartok) [inline hazelcast-jet-distribution/src/root/release_notes.txt:76]: This was in fact a more generic, JSON handling problem, just more obvious in the context of CDC. Now, if that's worth mentioning or not, that's your call. Just saying.

**Comment by olukas** [comment]
verify

**Pull request merged**